tests/core/test_expectation_suite.py

```diff
@@ -215,6 +215,36 @@ def expectation(self) -> gxe.ExpectColumnValuesToBeInSet:
             result_format="BASIC",
         )
 
+    @pytest.mark.unit
+    def test_instantiate_suite_with_expectations(self, expectation):
+        context = Mock(spec=AbstractDataContext)
+        set_context(project=context)
+        columns = ["a", "b", "c", "d", "e"]
+        expectations = [
+            expectation.copy(update={"column": column}) for column in columns
+        ]
+        suite = ExpectationSuite(
+            name=self.expectation_suite_name, expectations=expectations
+        )
+        assert suite.expectations == expectations
+
+    @pytest.mark.unit
+    def test_instantiate_suite_fails_with_expectations_with_ids(self, expectation):
+        context = Mock(spec=AbstractDataContext)
+        set_context(project=context)
+        columns = ["a", "b", "c", "d", "e"]
+        expectations = [
+            expectation.copy(update={"column": column, "id": uuid4()})
+            for column in columns
+        ]
+        with pytest.raises(
+            ValueError,
+            match="Expectations in parameter `expectations` must not belong to another ExpectationSuite.",
+        ):
+            ExpectationSuite(
+                name=self.expectation_suite_name, expectations=expectations
+            )
+
```

Comment:
If we need the runtime check about mixed types, we should have a test. But do we actually want to enforce that, or can we just handle mixed types.